src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -663,12 +663,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                       target,
                                                       diagnostics)
 
+                        Dim propertyAccess = TryCast(target, BoundPropertyAccess)
+
+                        If propertyAccess IsNot Nothing Then
+                            Debug.Assert(propertyAccess.AccessKind = PropertyAccessKind.Unknown)
+                            ' See if we can reclassify access as writable given that this is an object initializer.
+                            ' This is needed to accommodate init-only properties.
+                            If propertyAccess.AccessKind <> PropertyAccessKind.Get AndAlso Not propertyAccess.IsWriteable AndAlso
+                               propertyAccess.PropertySymbol.IsWritable(propertyAccess.ReceiverOpt, Me, isKnownTargetOfObjectMemberInintializer:=True) Then
+
+                                propertyAccess = propertyAccess.Update(propertyAccess.PropertySymbol, propertyAccess.PropertyGroupOpt, propertyAccess.AccessKind, isWriteable:=True,
+                                                                       propertyAccess.IsLValue, propertyAccess.ReceiverOpt, propertyAccess.Arguments, propertyAccess.DefaultArguments,
+                                                                       propertyAccess.Type)
+                                target = propertyAccess
```

Comment:
Do we need to check LangVersion here?

Reply:
If we don't, a comment as to why would be appreciated.

---
In reply to: [557618151](https://github.com/dotnet/roslyn/pull/50414#discussion_r557618151) [](ancestors = 557618151)

Reply:
> If we don't, a comment as to why would be appreciated.

We are simply deciding if the property is writable or not, the property is writable or not writable regardless of language version. I don't think any comments about why we don't do something here is necessary, we are not doing a lot of things here.

---
In reply to: [557618498](https://github.com/dotnet/roslyn/pull/50414#discussion_r557618498) [](ancestors = 557618498,557618151)